### PR TITLE
fix(sunburst): `radial` rotation label might upside down

### LINKED ISSUE #18023: [Bug] [Sunburst] Uses of startAngle provoke labels display bug
### Version

5.4.0

### Link to Minimal Reproduction

https://codesandbox.io/s/882hov

### Steps to Reproduce

1. Take the sunburst example : https://echarts.apache.org/examples/en/editor.html?c=sunburst-simple
2. Define a startAngle property like this :

option = {
&nbsp;&nbsp;&nbsp;&nbsp;series: {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;type: 'sunburst',
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;startAngle: 180, // <--------------------------
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;data: data,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;radius: \[0, '90%'\],
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;label: {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;rotate: 'radial'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;}
};

### Current Behavior

Label with rotate value to 'radial' are upside down
image

### Expected Behavior

Label in a good position

### Environment

```markdown
- OS: Windows 10
- Browser: Brave Version 1.46.133 Chromium: 108.0.5359.71
- Framework: Angular 14.2.2
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by susiwen8**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

`sunbrust` label might upside down under `radial` rotation

### Fixed issues

- Close #18023 

### Before: What was the problem?

Screenshot 2023-02-04 at 15 31 48

### After: How does it behave after the fixing?

Screenshot 2023-02-04 at 15 31 36

## Document Info

One of the following should be checked.

- [ ] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [x] The document changes have been made in https://github.com/apache/echarts-doc/pull/337

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

`sunburst-label.html`

### Merging options

- [x] Please squash the commits into a single one when merging.

### CODE REVIEW

**Review by Ovilia — CHANGES_REQUESTED** [review]
I forgot why `startAngle` is not provided in the document, whether it's a bug or intentionally. Anyway, if this issue is to be accepted, we should also update the document.
I think it should be a good idea to use `dat.gui` to dynamically change the `startAngle` to check it works well with all situations. We also may need to restrict the range of valid `startAngle` in the document.

**Comment by susiwen8** [comment]
> I forgot why `startAngle` is not provided in the document, whether it's a bug or intentionally. Anyway, if this issue is to be accepted, we should also update the document. I think it should be a good idea to use `dat.gui` to dynamically change the `startAngle` to check it works well with all situations. We also may need to restrict the range of valid `startAngle` in the document.

Ok, I will update it

**Comment by susiwen8** [comment]
@Ovilia Test case has updated, thanks to your suggestion about `dat.gui`, more strange case was exposed.
Also document for `startAngle` was added.

**Comment by Ovilia** [comment]
LGTM but not sure why the lint CI failed. @susiwen8 Can you have a double-check?

**Comment by susiwen8** [comment]
@Ovilia Ci has passed

File: test/sunburst-label.html [hunk 1]
```diff
@@ -112,6 +114,120 @@
...
+            function update() {
+                chart.setOption({
+                    series: [{
+                        startAngle: config.startAngle
+                    }]
+                });
+            }
+
+            var gui = new dat.GUI();
+                gui.add(config, 'startAngle', 0, 360)
```
Comment (plainheart) [inline test/sunburst-label.html:226]: As per https://github.com/apache/echarts/issues/18023#issuecomment-1337952610, it's better to also add `clockwise`.

File: src/chart/sunburst/SunburstPiece.ts [hunk 2]
```diff
@@ -251,8 +252,8 @@ class SunburstPiece extends graphic.Sector {
             const rotateType = getLabelAttr(labelStateModel, 'rotate');
             let rotate = 0;
             if (rotateType === 'radial') {
-                rotate = -midAngle;
-                if (rotate < -Math.PI / 2) {
+                rotate = normalizeRadian(-midAngle);
+                if (((rotate > Math.PI / 2 && rotate <= Math.PI) || (rotate > Math.PI && rotate <= Math.PI * 1.5))) {
```
Comment (plainheart) [inline src/chart/sunburst/SunburstPiece.ts:256]: It seems these two conditions can be merged into one condition. WDYT?

```ts
rotate > Math.PI / 2 && rotate <= Math.PI * 1.5
```

Comment (susiwen8) [reply src/chart/sunburst/SunburstPiece.ts:256]: Yeah, you are right

File: test/sunburst-label.html (hunk 1 above)
Comment (susiwen8) [reply test/sunburst-label.html:226]: `clockwise` has added

**Review by plainheart — APPROVED** [review]
LGTM

**Comment by susiwen8** [comment]
@Ovilia Please take a look

**Comment by Ovilia** [comment]
A little unexpected thing is that when the rotation of a text is changed by 360° (e.g., from 1 to 359), it has an animation of rotating the text. But I think we can ignore this for now and see if it's necessary to be fixed if some developers bring this up.

**Pull request merged**